core/src/main/kotlin/in/specmatic/core/git/GitOperations.kt

```diff
@@ -34,9 +34,20 @@ fun clone(workingDirectory: File, gitRepo: GitRepo): File {
     return cloneDirectory
 }
 
+fun checkout(workingDirectory: File, branchName: String) {
+    logger.log("Checking out branch: ${branchName}")
+    try {
+        SystemGit(workingDirectory.path).checkout(branchName);
+    } catch(exception: Exception) {
+        logger.debug("Could not checkout branch ${branchName}")
+        logger.debug(exception.localizedMessage ?: exception.message ?: "")
+        logger.debug(exception.stackTraceToString())
```

Comment:
Ideally would be helpful to exit the clone process by throwing a ```ContractException``` so that we exit and provide feedback to the user that branch was not available.